server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -232,8 +233,8 @@ public void cleanupExpiredPools() {
                     subAdapter.deleteSubscription(sub);
                 }
             }
-            deletePool(p);
         }
+        deletePools(toDelete);
```

Comment:
Ya, deletePools works itself into batch deleting everything. I think in this case, it was just something that was missed.